### PR TITLE
Feature/refactor for clip plane

### PR DESCRIPTION
**Pull request by toloudis**
This is a followup refactor on top of #225 

The goal of this change set is to finish decoupling selection from manipulation, and prepare the code to have the clip plane added.

Key changes:
* remove some special-case code for light source manipulation. For example, renaming AREA_LIGHT_MODE to MANIPULATOR_MODE.
* light source (or any object) must now be explicitly selected (in code) before it can be rotated
* move Transform3d out of Object3d and into MathUtils. Improve Transform3d::getMatrix
* get ready to allow both rotation and translation modes 

The next PR will actually add user clipping plane

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +20,7 @@ project(
 )
 
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (toloudis) [inline CMakeLists.txt:23]: this is purely an improvement for vscode performance issues. Has no effect on builds.

File: pyrightconfig.json
Comment (toloudis) [inline pyrightconfig.json:1]: This file is added purely for vscode performance reasons and has no effect on any builds or behavior of agave.

File: renderlib/renderlib.h
Comment (toloudis) [inline renderlib/renderlib.h:1]: The includes for some of the Qt classes in renderlib were moved around to make the builds a tiny bit faster. Qt includes can be a bit more heavyweight.

**Review by ShrimpCryptid — APPROVED** [review]
Had just the one question about how the emit/selection mechanism works, thank you for answering that! LGTM

**Review by interim17 — APPROVED** [review]
LGTM

**Pull request merged**